core/dbt/contracts/graph/unparsed.py

```diff
@@ -429,13 +431,34 @@ class UnparsedExposure(dbtClassMixin, Replaceable):
     type: ExposureType
     owner: ExposureOwner
     description: str = ""
+    label: Optional[str] = None
     maturity: Optional[MaturityType] = None
     meta: Dict[str, Any] = field(default_factory=dict)
     tags: List[str] = field(default_factory=list)
     url: Optional[str] = None
     depends_on: List[str] = field(default_factory=list)
     config: Dict[str, Any] = field(default_factory=dict)
 
+    @classmethod
+    def validate(cls, data):
+        super(UnparsedExposure, cls).validate(data)
+        # TODO: what do we actually want to rules to be?
+        if "name" in data:
+            errors = []
+            if " " in data["name"]:
+                errors.append("cannot contain spaces")
+            if len(data["name"]) > 126:
+                errors.append("cannot contain more than 126 characters")
+            if not (re.match(r"^[A-Za-z]", data["name"])):
+                errors.append("must begin with a letter")
+            if not (re.match(r"[\w-]+$", data["name"])):
+                errors.append("must contain only letters, numbers and underscores")
+
+            if errors:
+                raise ParsingException(
+                    f"Exposure name '{data['name']}' is invalid.  It {', '.join(e for e in errors)}"
+                )
```

Comment:
Could we actually make this `deprecations.warn` instead of `ParsingException`? I think there are many users (including [this integration](https://hightouch.com/blog/introducing-dbt-exposures-for-hightouch/)) who will be in violation of the new rule.

I can provide copy for the deprecation message, but the big idea is: "Hey! We just added support for a human-friendly `label` propertyon exposures. We see that {exposure} has some disallowed characters in its name — why not turn that into label, and then come up with a simpler name?"